components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java

```diff
@@ -78,6 +86,11 @@ public void handleEvent(Event event) throws IdentityEventException {
                 .EventProperty.CONTEXT);
         try {
             if (request == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("HttpServletRequest object is null. Hence getting the session related information from " +
+                            "event and revoking the access tokens bound to sso-session binding");
+                }
+                revokeAccessTokensMappedForSessions(event);
```

Comment:
add a audit log when session to token mapping  is revoked

Reply:
We have audit logs for each oauth related events in this repo https://github.com/wso2-extensions/identity-data-publisher-oauth/tree/02681715805e8d6375cee67c289e502f0df9ed43/components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener. We have token revocationaudit logger as well. So that can have this logging part. We don't need the explicitly handle it

Reply:
Got to know that the audit logs from these components don't have detailed context information such as action/event what caused the token revocation. Check the possibility of adding an audit log with the information about the session termination event.

Reply:
tracked with https://github.com/wso2/product-is/issues/10843